### PR TITLE
new mac address value object

### PR DESCRIPTION
**Pull request by marcosh**

### CODE REVIEW

File: src/Identity/MacAddress.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+use ValueObjects\StringLiteral\StringLiteral;
+
+class MacAddress extends StringLiteral
+{
+    /**
+     * Returns a MacAddress
+     *
+     * @param string $value
+     */
+    public function __construct($value)
```
Comment (andrzejkupczyk) [inline src/Identity/MacAddress.php]: I think that `$value` should be a StringLiteral OR it should call parent's constructor at first.
Consider this case:
```php
new MacAddress(array('1A-2b-3C-4d-5E-6f'));
```

File: src/Identity/MacAddress.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        if (0 === \preg_match('/^(?:[0-9a-f]{2}([-:]))(?:[0-9a-f]{2}\1){4}[0-9a-f]{2}$/i', $value)) {
```
Comment (andrzejkupczyk) [inline src/Identity/MacAddress.php]: Why not use the `FILTER_VALIDATE_MAC` filter? Just curious :)

Comment (marcosh) [inline src/Identity/MacAddress.php]: Thanks for the hint! I didn't know about it.

What do you think about rewriting the constructor as

    public function __construct($value)
    {
        $filteredValue = filter_var($value, FILTER_VALIDATE_MAC);

        if ($filteredValue === false) {
            throw new InvalidNativeArgumentException($value, array('string (valid Mac address)'));
        }

        $this->value = $filteredValue;
    }

similarly to what it is done for `EmailAddress`?

Comment (andrzejkupczyk) [inline src/Identity/MacAddress.php]: This is fine as it resolves https://github.com/nicolopignatelli/valueobjects/pull/71#discussion_r96137464 too :+1:

Comment (marcosh) [inline src/Identity/MacAddress.php]: I updated the pull request

**Comment by nicolopignatelli** [comment]
```php(new MacAddress('1A-2b-3C-4d-5E-6f'))->sameValueAs(new MacAddress('1A:2b:3C:4d:5E:6f')) // false
```

is it intended?

**Comment by marcosh** [comment]
@nicolopignatelli no, wan't intended; thanks for pointing it out.

I updated the PR solving the issue. Still at this point I'm not 100% sure if this is fine as it is or if it would be better to treat a mac address as six hexadecimal numbers, storing it internally that way (instead of a string) and exposing different constructors and visualization methods for the various formats. What do you think?

**Comment by nicolopignatelli** [comment]
I would go for the more generic option (your second one).

**Comment by marcosh** [comment]
@nicolopignatelli I updated the PR using a numeric internal representation

File: tests/Identity/MacAddressTest.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    public function testToStringWithDot()
+    {
+        $macAddress = MacAddress::fromString('ffff.ffff.ffff');
+
+        $this->assertEquals('ffff.ffff.ffff', $macAddress->toStringWithDot());
+    }
+
+    public function testSomething()
```
Comment (nicolopignatelli) [inline tests/Identity/MacAddressTest.php]: what are we testing here?

Comment (marcosh) [inline tests/Identity/MacAddressTest.php]: sorry, that shouldn't have been there...

**Comment by nicolopignatelli** [comment]
thank you

**Pull request merged**